src/modules/rppi_validate.hpp

```diff
@@ -336,4 +336,27 @@ inline void get_dstBatchIndex(rpp::Handle& handle, unsigned int channel, RppiChn
 }
 
 #endif // GPU_SUPPORT
+
+inline int check_roi_out_of_bounds(RpptROIPtr roiPtrImage, RpptDescPtr srcDescPtr, RpptRoiType type)
+{
+    int x, y, w, h;
+    if (type == RpptRoiType::XYWH)
+    {
+        x = (0 <= roiPtrImage->xywhROI.xy.x < srcDescPtr->w) ? roiPtrImage->xywhROI.xy.x : -1;
```

Comment:
why is this out of bound values set to -1?

Reply:
Here we check if point is within bounds as well as check if start offset / dimensions are valid i.e. >0, thus setting this to true / false will only check if value is out of bound and not validate negative x, y, width, height values.

Please correct me if I have understood this incorrectly.